service/src/main/java/tds/student/services/impl/RtsServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package tds.student.services.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+import tds.dll.common.rtspackage.IRtsPackageReader;
+import tds.dll.common.rtspackage.common.exception.RtsPackageReaderException;
+import tds.student.RtsStudentPackageAttribute;
+import tds.student.repositories.RtsStudentPackageQueryRepository;
+import tds.student.services.RtsService;
+
+@Service
+public class RtsServiceImpl implements RtsService {
+    private final RtsStudentPackageQueryRepository rtsStudentPackageQueryRepository;
+    private final IRtsPackageReader packageReader;
+
+    @Autowired
+    public RtsServiceImpl(RtsStudentPackageQueryRepository rtsStudentPackageQueryRepository, IRtsPackageReader packageReader) {
+        this.rtsStudentPackageQueryRepository = rtsStudentPackageQueryRepository;
+        this.packageReader = packageReader;
+    }
+
+    @Override
+    public Optional<RtsStudentPackageAttribute> findRtsStudentPackageAttribute(String clientName, long studentId, String attributeName) {
+        Optional<byte[]> maybePackage = rtsStudentPackageQueryRepository.findRtsStudentPackage(clientName, studentId);
```

Comment:
I think I assumed that most places would be querying more than a single value.  If that is not the case then we can table it for sure.